tests/test_search.py

```diff
@@ -88,6 +89,13 @@ def test_search_rotate(self):
         self.assertEqual(8, search_rotate_recur(array, 0, 11, 5))
         self.assertEqual(-1, search_rotate_recur(array, 0, 11, 9))
 
+    def test_jump_search(self):
+        array = [1, 2, 3, 3, 3, 3, 4, 4, 4, 4, 5, 6]
+        self.assertEqual(10, jump_search(array, 5))
+        self.assertEqual(11, jump_search(array, 6))
+        self.assertEqual(-1, jump_search(array, 7))
+        self.assertEqual(-1, jump_search(array, -1))
```

Comment:
What does `jump_search(array, 3)` return here? it can be any of the `2, 3, 4, 5`.
I think it depends on the block_size (and hence array size), in the case of repeated elements.

Hence, this algorithm should only be used when we are fine with any index in case of duplicate elements. I think we should include a small note about this too in the docstring.